plugins/hdf/io/Hdf5Handler.cpp

```diff
@@ -0,0 +1,199 @@
+/******************************************************************************
+* Copyright (c) 2014, Connor Manning, connor@hobu.co
+*
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following
+* conditions are met:
+*
+*     * Redistributions of source code must retain the above copyright
+*       notice, this list of conditions and the following disclaimer.
+*     * Redistributions in binary form must reproduce the above copyright
+*       notice, this list of conditions and the following disclaimer in
+*       the documentation and/or other materials provided
+*       with the distribution.
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "Hdf5Handler.hpp"
+#include <pdal/util/FileUtils.hpp>
+#include <pdal/pdal_types.hpp>
+#include <pdal/Dimension.hpp>
+
+namespace pdal
+{
+
+using namespace hdf5;
+
+void Handler::setLog(pdal::LogPtr log) {
+    m_logger = log;
+}
+
+
+void Handler::initialize(
+        const std::string& filename,
+        const std::map<std::string,std::string>& map)
+{
+    try
+    {
+        m_h5File.reset(new H5::H5File(filename, H5F_ACC_RDONLY));
+    }
+    catch (const H5::FileIException&)
+    {
+        throw pdal_error("Could not open HDF5 file '" + filename + "'.");
+    }
+
+    // Create our vector of dimensions and associated data
+    for( auto const& entry : map) {
+        std::string const& dimName = entry.first;
+        std::string const& datasetName = entry.second;
+        m_dimInfos.emplace_back(DimInfo(dimName, datasetName, m_h5File.get()));
+    }
+
+    // Check that all dimensions have equal lengths
+    m_numPoints = m_dimInfos.at(0).getNumPoints();
+    for( DimInfo& info : m_dimInfos) {
+        if(m_numPoints != info.getNumPoints()) {
+            throw pdal_error("All given datasets must have the same length");
```

Comment:
It would be helpful to tell the user which dimension they gave that doesn't have the same count as the others.